seed/campaigns.ts

```diff
@@ -1,59 +1,143 @@
+  data: PrismaJson.PathToVictoryData
+}
 
 export default async function seedCampaigns(
   prisma: PrismaClient,
   existingUsers: User[],
 ) {
-  const fakeP2Vs: any[] = []
-  const fakeUpdateHistory: any[] = []
-
+  const fakeP2Vs: FakeP2V[] = []
+  const fakeUpdateHistory: CampaignUpdateHistory[] = []
   const campaignIds: number[] = []
 
-  for (let i = 0; i < NUM_CAMPAIGNS; i++) {
-    let user = existingUsers[i]
-    if (!user) {
-      const userData = userFactory()
-      user = await prisma.user.create({
-        data: {
-          ...userData,
-          metaData:
-            userData.metaData !== null ? userData.metaData : Prisma.JsonNull,
-        },
-      })
-    }
-    const campaign = await prisma.campaign.create({
-      data: campaignFactory({
-        userId: user.id,
-        slug: buildSlug(getFullName(user)),
-      }),
-    })
-
-    campaignIds.push(campaign.id)
-    fakeP2Vs.push(pathToVictoryFactory({ campaignId: campaign.id }))
+  const loopLength = Math.max(FIXED_CAMPAIGNS.length, NUM_GENERATED_CAMPAIGNS)
 
-    for (let j = 0; j < NUM_UPDATE_HISTORY; j++) {
-      fakeUpdateHistory.push(
-        campaignUpdateHistoryFactory({
-          campaignId: campaign.id,
-          userId: user.id,
-        }),
+  for (let i = 0; i < loopLength; i++) {
+    if (i < FIXED_CAMPAIGNS.length) {
+      const creationData = await createCampaignAndUser(
+        existingUsers,
+        prisma,
+        FIXED_CAMPAIGNS[i],
       )
+      const { campaignId, fakeP2V, fakeUpdateHistory } = creationData
+
+      campaignIds.push(campaignId)
+      fakeP2Vs.push(fakeP2V)
+      fakeUpdateHistory.push(...fakeUpdateHistory)
+    }
+    if (i < NUM_GENERATED_CAMPAIGNS) {
+      const creationData = await createCampaignAndUser(existingUsers, prisma)
+      const { campaignId, fakeP2V, fakeUpdateHistory } = creationData
+
+      campaignIds.push(campaignId)
+      fakeP2Vs.push(fakeP2V)
+      fakeUpdateHistory.push(...fakeUpdateHistory)
     }
   }
 
-  await prisma.pathToVictory.createMany({ data: fakeP2Vs })
+  //await prisma.pathToVictory.createMany({ data: fakeP2Vs })
   await prisma.campaignUpdateHistory.createMany({ data: fakeUpdateHistory })
 
   console.log(`Created ${campaignIds.length} campaigns`)
 
   return campaignIds
 }
+
+async function createCampaignAndUser(
+  existingUsers: User[],
+  prisma: PrismaClient,
+  fixedData?: Partial<Campaign>,
+): Promise<{
+  campaignId: number
+  fakeP2V: FakeP2V
+  fakeUpdateHistory: CampaignUpdateHistory[]
+}> {
+  const user = await handleUserCreation(prisma, existingUsers)
+  let campaign: Campaign
+  if (fixedData) {
+    campaign = await prisma.campaign.create({
+      data: campaignFactory({
+        userId: user.id,
+        slug: buildSlug(getFullName(user)),
+        ...fixedData,
+      }),
+    })
+  } else {
+    campaign = await prisma.campaign.create({
+      data: campaignFactory({
+        userId: user.id,
+        slug: buildSlug(getFullName(user)),
+      }),
+    })
+  }
```

Comment:
You could just use a ternary here:

```suggestion
  const campaign =  await prisma.campaign.create({
      data: campaignFactory({
        userId: user.id,
        slug: buildSlug(getFullName(user)),
        ...(fixedData || {}),
      }),
    })
```